src/sentry/api/endpoints/organization_member_invite/details.py

```diff
@@ -140,7 +145,90 @@ def put(
 
         return Response(serialize(invited_member, request.user), status=200)
 
+    def _remove_invite_from_db(
+        self, request: Request, organization: Organization, invited_member: OrganizationMemberInvite
+    ) -> None:
+        audit_data = invited_member.get_audit_log_data()
+        event_name = "INVITE_REMOVE" if invited_member.invite_approved else "INVITE_REQUEST_REMOVE"
+
+        with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+            # also deletes the invite object via cascades
+            invited_member.organization_member.delete()
+
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=invited_member.id,
+                event=audit_log.get_event_id(event_name),
+                data=audit_data,
+            )
+
+    def _handle_deletion_by_member(
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
+        acting_member: OrganizationMember,
+    ) -> Response:
+        # Members can only delete invitations that they sent
+        if invited_member.inviter_id != acting_member.user_id:
+            return Response({"detail": ERR_MEMBER_INVITE}, status=400)
+
+        self._remove_invite_from_db(
+            request=request, organization=organization, invited_member=invited_member
+        )
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
     def delete(
         self, request: Request, organization: Organization, invited_member: OrganizationMemberInvite
     ) -> Response:
-        raise NotImplementedError
+        # with superuser read write separation, superuser read cannot hit this endpoint
+        # so we can keep this as is_active_superuser
+        if request.user.is_authenticated and not is_active_superuser(request):
```

Comment:
maybe check the user is authenticated separately and early return? they should always be authenticated to proceed